src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitRequestsService.kt

```diff
@@ -20,4 +25,39 @@ class VisitRequestsService(
 
     return visitRepository.getCountOfRequestedVisitsForPrison(prisonCode).toInt()
   }
+
+  fun getVisitRequestsForPrison(prisonCode: String): List<VisitRequestSummaryDto> {
+    val visitRequests = visitRepository.getRequestedVisitsForPrison(prisonCode)
+
+    val visitRequestSummaryList = mutableListOf<VisitRequestSummaryDto>()
+
+    for (visit in visitRequests) {
+      // To avoid failing the entire call, if we cannot get the prisoner details, swallow the exception and use placeholder values instead of name.
+      val prisoner = try {
+        prisonerService.getPrisoner(visit.prisonerId)
+      } catch (e: Exception) {
+        LOG.error("Unable to find prisoner ${visit.prisonerId}, using placeholder for visit request summary prisoner name", e)
+        null
+      }
```

Comment:
we could reduce the calls if we get prisonername by distinct prisonerids on all visits as the same prisoner might have multiple visits.